### PR TITLE
Update processing algorithms

### PR DESCRIPTION
**Pull request by Mtk112**
Added missing algorithms.
Modified project structure.
Added drafts subdirectory, where work in progress algorithms can be stored.
Moved feature importance, CBA, correlation matrix and covariance matrix algorithms under drafts.

### CODE REVIEW

File: eis_qgis_plugin/processing/algorithms/validation/plot_correlation_matrix.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    def initAlgorithm(self, config=None):
+        self.alg_parameters = ["matrix", "annotate", "cmap", "plot_title", "**kwargs", "out_raster"]
+
+        self.addParameter(
+            QgsProcessingParameterFile(
+                name=self.alg_parameters[0],
+                description="Correlation matrix.",
+            )
+        )
```
Comment (nmaarnio) [inline eis_qgis_plugin/processing/algorithms/validation/plot_correlation_matrix.py:30]: I'm not 100% sure, but I think this function is used for vector data, so you could change the input to that (this function call would actually both produce the correlation matrix and then plot it).

File: eis_qgis_plugin/processing/algorithms/vector_processing/cell_based_association.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            "add_name",
+            "add_buffer",
+            "output_vector"
+        ]
+
+        self.addParameter(
+            QgsProcessingParameterVectorLayer(
+                name=self.alg_parameters[0], description="Input vector"
+            )
+        )
```
Comment (nmaarnio) [inline eis_qgis_plugin/processing/algorithms/vector_processing/cell_based_association.py:36]: According to Toolkit, there could be multiple vector inputs.

File: eis_qgis_plugin/processing/algorithms/vector_processing/cell_based_association.py
Comment (nmaarnio) [inline eis_qgis_plugin/processing/algorithms/vector_processing/cell_based_association.py:1]: This is quite complicated to implement with the input definitions of Toolkit... Maybe just leave it as draft for now, and let's take a look at it later? So you could comment everything out in the file and leave it like that

**Review by nmaarnio — CHANGES_REQUESTED** [review]

File: eis_qgis_plugin/processing/algorithms/exploratory_analysis/feature_importance.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                description="Testing feature data (X)",
+            )
+        )
+
+        self.addParameter(
+            QgsProcessingParameterFile(
+                name=self.alg_parameters[2],
+                description="Testing feature data (Y)"
+            )
+        )
```
[the remark this answers is not in the document]
Comment (Mtk112) [reply eis_qgis_plugin/processing/algorithms/exploratory_analysis/feature_importance.py:53]: Changed these to QgsProcessingParameterRasterLayer, double checked that x_test on toolkit expects a single np.ndarray. Also, I moved this to drafts, more about drafts further down the comments.

File: eis_qgis_plugin/processing/algorithms/exploratory_analysis/statistical_tests.py
[the remark this answers is not in the document]
Comment (Mtk112) [reply eis_qgis_plugin/processing/algorithms/exploratory_analysis/statistical_tests.py:1]: My mistake, added all the functions as separate algorithms.

File: eis_qgis_plugin/processing/algorithms/transformations/one_hot_encoding.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+        self.addParameter(
+            QgsProcessingParameterString(
+                name=self.alg_parameters[1],
+                description="Target columns (for vector)",
+                multiLine=True,
+                optional=True,
+                defaultValue=None
+            )
+        )
```
[the remark this answers is not in the document]
Comment (Mtk112) [reply eis_qgis_plugin/processing/algorithms/transformations/one_hot_encoding.py:54]: Changed parameter type to Field. I added clarification to the description.

File: eis_qgis_plugin/processing/algorithms/vector_processing/cell_based_association.py
Comment (Mtk112) [reply eis_qgis_plugin/processing/algorithms/vector_processing/cell_based_association.py:1]: I created a new drafts directory under algorithms, as just commenting out the files causes issues with building the plugin. I moved CBA and Feature importance under the drafts directory to mark them as work in progress.

**Pull request merged**